analyses/infercnv-consensus-cell-type/template-notebooks/ewings-infercnv-results.Rmd

```diff
@@ -0,0 +1,536 @@
+    title = "Immune-subset reference"
+  ) +
+  theme(legend.position = "bottom")
+
+wrap_plots(all_plot, subset_plot) +
+  plot_annotation(title = "Total CNVs per cell")
+```
+
+### CNVs across cell types
+
+Next, we'll look at the per-cell distributions of CNVs across cell types.
+Specifically, we'll look at both the cell types from the `cell-type-consensus` module and the `cell-type-ewings` module separately.
+
+Ideally, we'll see that Unknown cells from `cell-type-consensus` and cells labeled as tumor in `cell-type-ewings` will have relatively higher CNV scores compared to other cell types.
+In addition, comparing distributions between `all-immune` and `subset-immune` may be helpful to identify the optimal normal reference strategy.
+
+```{r fig.height=10, fig.width=12, message=FALSE, warning=FALSE}
+# drop cells from other libraries
+query_has_cnv_df <- has_cnv_df |>
+  dplyr::filter(library_id == params$library_id)
+
+consensus_violin <- ggplot(query_has_cnv_df) +
+  aes(
+    x = consensus_annotation,
+    y = total_cnv_per_cell,
+    fill = normal_reference
+  ) +
+  geom_violin(scale = "width") +
+  scale_fill_brewer(palette = "Pastel2") +
+  labs(title = "Annotations from cell-type-consensus") +
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+
+ewings_violin <- ggplot(query_has_cnv_df) +
+  aes(
+    x = ewing_annotation,
+    y = total_cnv_per_cell,
+    fill = normal_reference
+  ) +
+  geom_violin(scale = "width") +
+  scale_fill_brewer(palette = "Pastel2") +
+  labs(title = "Annotations from cell-type-ewings") +
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+
+wrap_plots(consensus_violin, ewings_violin, ncol = 1, guides = "collect") +
+  plot_annotation(title = "CNV scores across cell types") &
+  theme(legend.position = "bottom")
+```
+
+
+
+## UMAPs
+
+Next, we'll visualize the total CNV counts as a UMAP.
+
+First, as a guide to interpret the UMAP, let's color by the top 10 `cell-type-ewings` cell types.
+
+```{r, fig.width = 6}
+ggplot(umap_df) +
+  aes(x = UMAP1, y = UMAP2, color = forcats::fct_lump(ewing_annotation, 8)) +
+  geom_point(size = 0.1, alpha = 0.3) +
+  labs(color = "ewings annotation") +
+  guides(color = guide_legend(override.aes = list(size = 2, alpha = 1))) +
+  umap_theme
+```
+
+Next, the total CNVs per cell; ideally, higher values will overlap with regions in the UMAP containing putative tumor cells.
+
+```{r fig.width=10, warning = FALSE}
+# noting we expect warnings with this plot since there's no UMAP coordinates
+# for cells from other libraries in the reference
+ggplot(has_cnv_df) +
+  aes(x = UMAP1, y = UMAP2, color = total_cnv_per_cell) +
+  geom_point(size = 0.1, alpha = 0.3) +
+  scale_color_viridis_c() +
+  facet_wrap(vars(normal_reference)) +
+  labs(title = "Total CNV per cell") +
+  umap_theme
+```
+
+## Compare references directly
+
+We can also look at CNV predictions directly between references using a point-density scatterplot, with the Spearman correlation shown:
```

Comment:
I don't know what I get from this plot? I do think we might want to compare the two references directly, but I'm not sure this is the way to go. I'm interested in what the CNV levels look like in tumor cells across references which you already have shown me above. I also want to know if we use the CNV profiles to classify tumor cells how accurate are we and which reference does a better job. That requires some figuring out how we actually use inferCNV to classify tumor cells and I think might be outside the scope of this notebook.

Reply:
I can remove this plot! There's plenty else anyways 😬